path_oidc.go

```diff
@@ -106,6 +153,22 @@ func (b *jwtAuthBackend) pathCallback(ctx context.Context, req *logical.Request,
 		return logical.ErrorResponse(errLoginFailed + " Could not load configuration"), nil
 	}
 
+	stateID := d.Get("state").(string)
+
+	state := b.verifyState(stateID)
+	if state == nil {
+		return logical.ErrorResponse(errLoginFailed + " Expired or missing OAuth state."), nil
+	}
+
+	roleName := state.rolename
+	role, err := b.role(ctx, req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse(errLoginFailed + " Role could not be found"), nil
```

Comment:
I actually asked core folks today if we should return `nil, nil` or an error response.  Seems like `nil, nil` was the preferred way.  Wondering if there's a special reason why we're returning a response here?

Reply:
Mainly so the UI can display something more helpful/specific. The OIDC flow has more steps and things that could fail or be misconfigured, so surfacing detailed errors was a goal.

<img width="503" alt="image" src="https://user-images.githubusercontent.com/211880/74471192-a6bf5d80-4e54-11ea-9d66-510119401116.png">